provider.md

```diff
@@ -0,0 +1,44 @@
+# Adding Providers
+This document will guide you through the process of creating a concrete resource converter that implements the `Provider`
+interface in the `i2gw` package.
+
+## Overview
+Each provider implementation in the `i2gw/providers` package is responsible for converting a provider specific `Ingress`
+and related resources (e.g istio VirtualService) into `Gateway API` resources.
+A provider must be able to read its custom resources, and convert them.
+
+## Prerequisites
+* Familiarity with Go programming language.
+* Basic understanding of Kubernetes and its custom resources.
+* A setup Go development environment.
+
+## Step-by-Step Implementation
+1. Add a new package under the `providers` package.
+2. Create a struct named `resourceReader` which implements the `CustomResourceReader` interface in a file named
+`resource_converter.go`.
+3. Create a struct named `converter` which implements the `ResourceConverter` interface in a file named `converter.go`.
+The implemented `ToGatewayAPI` function should simply call every registered `featureParser` function, one by one.
+Take a look at `ingressnginx/converter.go` for example.
```

Comment:
can this be split into sub bullets or separate bullets ?

Reply:
I think this is pretty atomic as it is, how would you like to see it separated?